src/AppBundle/Security/ApiKeyAuthenticator.php

```diff
@@ -11,27 +13,37 @@
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\HttpKernel\Exception\HttpException;
 
 class ApiKeyAuthenticator extends AbstractGuardAuthenticator
 {
     /**
      * @var ManagerRegistry
      */
     private $registry;
-
     /**
-     * @param ManagerRegistry $registry
+     * @var Logger
      */
-    public function __construct(ManagerRegistry $registry)
+    private $logger;
+
+    public function __construct(ManagerRegistry $registry, Logger $logger)
     {
         $this->registry = $registry;
+        $this->logger = $logger;
     }
 
     /**
-     * @inheritdoc
+     * {@inheritdoc}
      */
     public function getCredentials(Request $request)
     {
+        $client = $this->registry->getRepository('AppBundle:Client')
```

Comment:
you are violate Single responsibility principle. Use separate event listener with high priority to hit this condition.